enhancements/helm-mediatype.md

```diff
@@ -0,0 +1,338 @@
+---
+title: helm-mediatype
+authors:
+  - "@bowenislandsong"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-May-26
+last-updated: 2020-Jun-25
+status: provisional
+see-also:
+  - "./enhancements/csvless-bundles.md"
+---
+
+# helm-mediatype
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+ - Can OLM support Chart deprecation from a registry? [Deprecate Charts](https://helm.sh/docs/topics/charts/#deprecating-charts)?
+ - Can OLM support at least parts of the Chart [Hooks](https://helm.sh/docs/topics/charts_hooks/) functionalities such as pre/post-install/upgrade? (OLM does not have operator rollback, delete, and test features at this moment.)
+ - Should an operator package allow bundles of mixed media types and will upgrades between them sail smoothly?
+ - How to avoid OLM/Registry from the previous versions to install/add Helm charts?
+ - Can install mode for an operator be default to a value/inferred based on any information in the chart?
+ - Is it safe to assume that all [built-in objects](https://helm.sh/docs/chart_template_guide/builtin_objects/), including capabilities, are evaluated during chart render time and should not affect OLM operations?
+ - What would be a good CLI UX experience for users to put in chart settings/values on subscription/update?
+
+## Summary
+
+This enhancement adds support to accept `helm chart` media type for [Operator Bundle spec](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md), [operator-registry](https://github.com/operator-framework/operator-registry), and [OLM](https://github.com/operator-framework/operator-lifecycle-manager) to enable deployment/upgrading helm-based operators. This enhancement covers the design/UX for opm and OLM, and pipeline for building, adding, and installing helm-based operators that follow the [conventions](https://helm.sh/docs/chart_best_practices/conventions/) of `Helm+v3`.
```

Comment:
Operator-SDK allows users to create operators based on the helm-charts and it provides the features for them to work with as build the bundle. So, why it would be required?

Reply:
That requires the helm chart developer to convert their chart into an operator bundle. IMO this would be more valuable if there was a way for OLM to consume helm charts directly rather than have a manual conversion take place before they could be consumed (which adding a bundle type as is described above doesn't accomplish)

Reply:
IMO it is not part of the domain of responsibility of OLM. OLM has the goal to distribute operators projects which can be done via the bundle. Also, will OLM generated the CRD's? will OLM generate a controller using helm native implementation to manage the charts?

Reply:
That's true today, but that's just because OLM has currently limited itself to deploying and installing operators. It seems reasonable for OLM to potentially branch out to other types of applications on a cluster outside the scope of the strict definition of operators defined by the operator framework's bundle format.

It may be that in the case of helm charts we can't easily define a UX that enables the templating system that helm defines. But at a high level one of the killer features that I see OLM provides is it's dependency resolution. Why should the dependency resolver be limited just to other bundles if there was a convenient entryway for non-operators (but still applications or kube objects) to be applied in addition to operators in the strictest sense of the world. We already allow users to add certain additional kube types to their bundles in OLM, why do those kube types need be included in the bundle as a yaml object itself?